presto-docs/src/main/sphinx/connector/hive.rst

```diff
@@ -559,6 +559,10 @@ Procedures
 
     Create an empty partition in the specified table.
 
+* ``system.msck(schema_name, table_name, add, drop)``
+
+    Check partitions in metastore and in file system. Repair the partitions(``add`` and ``drop``) if requested.
```

Comment:
I agree, the name `msck` is not descriptive, and having two booleans is confusing. Also, there are only three valid states, not four, since having both be `false` would do nothing.

What about having three functions:

* `add_partitions`
* `drop_partitions`
* `sync_partitions`

These are be similar to the `MSCK` options for Hive. I thought of some longer names, but it's hard to be  descriptive without making the names too long, and it seems good to have add/drop be symmetric.

For the docs, rather than saying "repair", something more direct like

> Add any partitions that exist on the file system but not in the metastore.

> Drop any partitions that exist in the metastore but not on the file system.

Reply:
Or, how about let's make it something like
``` 
system.repair_partitions(schema, table, mode)
```
where `mode` (`VARCHAR` type) is a one of `ADD`, `DROP` or `SYNC`.

Reply:
I dislike the name “repair” since it implies we are fixing them (because they are damaged?), when in reality we are synchronizing the metadata with the file system.

Although if we want to stick with the “repair” name, or any single name, your proposal seems good.

Reply:
Another idea, based on yours: `sync_partition_metadata` with modes `ADD`, `DROP`, `FULL`.

Reply:
`sync_partition_metadata` looks good. I'll update the PR with the new name.